src/main/java/uk/gov/pay/ledger/transaction/dao/TransactionDao.java

```diff
@@ -213,6 +213,18 @@ public class TransactionDao {
                     "agreement_id = EXCLUDED.agreement_id " +
                     "WHERE EXCLUDED.event_count >= transaction.event_count;";
 
+    private static final String REDACT_PII_FROM_TRANSACTIONS =
+            "UPDATE transaction t " +
+                    "set reference = '<REDACTED>'," +
+                    "  cardholder_name = (case when cardholder_name IS NOT NULL then '<REDACTED>' end), " +
+                    "  email = (case when email IS NOT NULL then '<REDACTED>' end), " +
```

Comment:
What happens to the update if the cardholder_name and/or email is null?

Reply:
They will stay null in that case. This test case should cover it https://github.com/alphagov/pay-ledger/pull/2894/files#diff-dcd3f9f3fa9bb7ad5ba913ef2f3e3b6c846e994cf7bdd9690cda980ed3410c29R571-R573